### PR TITLE
feat: add more comparison functions, docs

### PR DESCRIPTION
**Pull request by nilslice**
Adds and tests the following functions: 

```ts
 // assert that `x` is greater than `y`, naming the assertion with `msg`, which will be used as a label in the CLI runner.
  static assertGreaterThan(msg: string, x: any, y: any) { ... }

  // assert that `x` is greater than or equal to `y`, naming the assertion with `msg`, which will be used as a label in the CLI runner.
  static assertGreaterThanOrEqualTo(msg: string, x: any, y: any) { ... }

  // assert that `x` is less than `y`, naming the assertion with `msg`, which will be used as a label in the CLI runner.
  static assertLessThan(msg: string, x: any, y: any) { ... }

  // assert that `x` is less than or equal to `y`, naming the assertion with `msg`, which will be used as a label in the CLI runner.
  static assertLessThanOrEqualTo(msg: string, x: any, y: any) { ... }
```

Notice the `x` and `y` arguments type changes to `any` here. I learned that `unknown` cannot be compared aside for equality, and couldn't come up with a good way to cast `unknown` to something reliably comparable. 

I think it's ok?

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -138,4 +138,32 @@ export class Test {
     stack.trim();
     Test.assert(msg, x !== y, `Expected ${x} !== ${y}\n${stack}`);
   }
+
+  // assert that `x` is greater than `y`, naming the assertion with `msg`, which will be used as a label in the CLI runner.
+  static assertGreaterThan(msg: string, x: any, y: any) {
+    const stack = new Error().stack || "";
+    stack.trim();
```
Comment (zshipko) [inline index.ts]: Might be time to make this a function

**Review by zshipko — APPROVED** [review]
Looks good, just one nit that you can feel free to ignore if you want

File: index.ts [hunk 2]
```diff
@@ -138,4 +138,32 @@ export class Test {
...
+    Test.assert(msg, x < y, `Expected ${x} < ${y}\n${stack}`);
+  }
+
+  // assert that `x` is less than or equal to `y`, naming the assertion with `msg`, which will be used as a label in the CLI runner.
+  static assertLessThanOrEqualTo(msg: string, x: any, y: any) {
+    const stack = new Error().stack || "";
+    stack.trim();
+    Test.assert(msg, x <= y, `Expected ${x} <= ${y}\n${stack}`);
+  }
```
Comment (chrisdickinson) [inline index.ts:156]: I wonder if we should converge on [`node:assert` types](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/assert.d.ts) for test assertions? It's pretty well-supported across platforms and browsers.

**Review by chrisdickinson — APPROVED** [review]
LGTM!

File: index.ts (hunk 2 above)
Comment (nilslice) [reply index.ts:156]: @chrisdickinson probably a good idea! I think as long as they still wrap our own internal functions so the outputs are consistent across test runs, making the most idiomatic API for a language ecosystem is the right move. 

I will merge as-is, and we can re-visit (perhaps when/if you'd like to take it on... :smile: )

File: index.ts (hunk 1 above)
Comment (nilslice) [reply index.ts]: Done! (also pruned the stack so this new extracted function isn't included in the trace for every failure)

**Pull request merged**